### PR TITLE
2604 Get raw loss values of hybrid loss

### LINKED ISSUE #2604: Add support to access raw losses of hybrid loss
**Is your feature request related to a problem? Please describe.**
Feature request from Clara users:

with hybrid loss we need to display each loss separately so that user can calibrate the weights

loses are on different scale to begin with dice range is 0-1  while cross entropy is 0-inf and depends on the size of the volume 
```
need to have out put similar to the one in TF below
Epoch: 1/5, Iter: 1/11 [=                   ]  train_ce_loss: 0.3277  train_dice_loss: 0.9091  train_loss: 4.1865  time: 16.89s        
Epoch: 1/5, Iter: 2/11 [===                 ]  train_ce_loss: 0.2897  train_dice_loss: 0.8875  train_loss: 3.7847  time: 12.38s 
```

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2604 .

### Description
This PR added support to get the raw loss values of hybrid loss function.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/losses/dice.py [hunk 1]
```diff
@@ -690,9 +696,19 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
             target = torch.squeeze(target, dim=1)
         target = target.long()
         ce_loss = self.cross_entropy(input, target)
+        self.loss_details["ce_loss"] = ce_loss.detach()
         total_loss: torch.Tensor = self.lambda_dice * dice_loss + self.lambda_ce * ce_loss
+        self.loss_details["total_loss"] = total_loss.detach()
+
         return total_loss
 
+    def get_loss_details(self) -> Dict[str, torch.Tensor]:
```
Comment (wyli) [inline monai/losses/dice.py:705]: I think if we have `print(my_loss.get_loss_details())` it'll cause the gpu tensor to be synced to cpu and have an impact on the speed? not sure if it's a good approach to fetch the intermediate cuda tensor results

Comment (Nic-Ma) [reply monai/losses/dice.py:705]: Hi @wyli ,

Here I just **detach** the loss tensors from graph, not moving it to CPU. Do you mean the `print()` function will potentially move the tensor to CPU? If so, it will impact the speed if you call the `print()` function?
```py
>>> import torch
>>> 
>>> a = torch.tensor(0.5, device="cuda:0")
>>> 
>>> b = a.detach()
>>> 
>>> print(b)
tensor(0.5000, device='cuda:0')
```

Thanks.

Comment (wyli) [reply monai/losses/dice.py:705]: I think it's true according to https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#avoid-unnecessary-cpu-gpu-synchronization

Comment (Nic-Ma) [reply monai/losses/dice.py:705]: Hi @wyli ,

So in that case, if users want a better perf, they just don't call the `print(my_loss.get_loss_details())`?
Maybe let me enhance the doc-string to make it clear?

Thanks.

Comment (Nic-Ma) [reply monai/losses/dice.py:705]: Users can use this `get_loss_details()` to tune the weights of hybrid loss as the feature request, if want the best perf of training, avoid printing the loss details?

Thanks.

Comment (Nic-Ma) [reply monai/losses/dice.py:705]: updated doc in the latest commit.

Thanks.

Comment (wyli) [reply monai/losses/dice.py:707]: I feel for debugging purposes the user can create a `DiceFocalLoss`, and call `DiceFocalLoss(...).dice(inp, target)`, and `DiceFocalLoss(...).focal(inp, target)` to get the internal values. Perhaps we improve `DiceCELoss` for the same?

Comment (Nic-Ma) [reply monai/losses/dice.py:707]: Hi @wyli ,

Thanks for your suggestion.
For workflow users, I think they can call `get_loss_details()` in a handler to print or debug instead of inheriting trainer and rewrite the `iteration()`?

Thanks.

Comment (wyli) [reply monai/losses/dice.py:707]: I think if we follow this idea, then for many other components, we would have to create temporary variables and add `get_*_details()` everywhere, for example in the networks, optimizers. Not sure if this is our recommended way to debug the components...

Comment (Nic-Ma) [reply monai/losses/dice.py:707]: @ericspod @rijobro , what do you guys think about the solution for this feature request? To tune the weights of hybrid losses. Maybe you have other ideas?

Thanks in advance.

Comment (Nic-Ma) [reply monai/losses/dice.py:707]: @ericspod @rijobro , if you guys have no more comments, I will update the PR to use separate `dice` and `ce` functions according to @wyli 's suggestion.

Thanks.

Comment (wyli) [reply monai/losses/dice.py:707]: perhaps it's a good opportunity to discuss the usability, we can gather more comments on this...

Comment (Nic-Ma) [reply monai/losses/dice.py]: Updated in the latest commit to provide the API to compute CE loss in `DiceCELoss`, so users can call `self.dice()` and `self.ce` for debug. Same as `self.focal()` as @wyli suggested.

Thanks.

**Review by ericspod — APPROVED** [review]

**Comment by ericspod** [comment]
I think this is a clean solution for this.
